xdem/spatialstats.py

```diff
@@ -1394,6 +1802,142 @@ def neff_hugonnet_approx(coords: np.ndarray, errors: np.ndarray, params_variogra
 
     return neff
 
+def number_effective_samples(area: float | int | VectorType | gpd.GeoDataFrame, params_variogram_model: pd.DataFrame,
+                             rasterize_resolution: RasterType | float = None, **kwargs) -> float:
+    """
+    Compute the number of effective samples, i.e. the number of uncorrelated samples, in an area accounting for spatial
+    correlations described by a sum of variogram models.
+
+    This function wraps two methods:
+
+    - A discretized integration method that provides the exact estimate for any shape of area using a double sum of
+        covariance. By default, this method is approximated using Equation 18 of Hugonnet et al. (2022), https://doi.org/10.1109/jstars.2022.3188922
+        to decrease computing times while preserving a good approximation.
+
+    - A continuous integration method that provides a conservative (i.e., slightly overestimated) value for a disk
+        area shape, based on a generalization of the approach of Rolstad et al. (2009), http://dx.doi.org/10.3189/002214309789470950.
+
+    By default, if a numeric value is passed for an area, the continuous method is used considering a disk shape. If a
+    vector is passed, the discretized method is computed on that shape. If the discretized method is used, a resolution
+    for rasterization is generally expected, otherwise is arbitrarily chosen as a fifth of the shortest correlation
+    range to ensure a sufficiently fine grid for propagation of the shortest range.
+
+    :param area: Area of interest either as a numeric value of surface in the same unit as the variogram ranges (will
+        assume a circular shape), or as a vector (shapefile) of the area
```

Comment:
Alternatively, it could be a boolean array, with pixels of interest set to True.

Reply:
But then there is no georeferencing, so we'd need an additional argument...
A "boolean Raster" would work.